evap/results/tests/test_views.py

```diff
@@ -473,6 +472,46 @@ def test_preview_with_rating_answers(self):
         url = f"/results/semester/{self.semester.id}/evaluation/{evaluation.id}"
         self.app.get(url, user=self.manager)
 
+    def test_unpublished_single_results_show_results(self):
+        """Regression test for #1621"""
+        # make regular evaluation with some answers
+        evaluation = baker.make(
+            Evaluation, state=Evaluation.State.REVIEWED, course=baker.make(Course, semester=self.semester)
+        )
+        questionnaire = baker.make(Questionnaire, type=Questionnaire.Type.TOP)
+        likert_question = baker.make(Question, type=Question.LIKERT, questionnaire=questionnaire, order=1)
+        evaluation.general_contribution.questionnaires.set([questionnaire])
+        participants = baker.make(UserProfile, _bulk_create=True, _quantity=20)
+        evaluation.participants.set(participants)
+        evaluation.voters.set(participants)
```

Comment:
could in theory directly be passed to the `baker.make` call for `evaluation`, or is there some related-objects-not-saved-blah issue I don't see? The objects should be stored and have their pk set here. Same for below (499-500)